### PR TITLE
fix continuing sync from subfolder

### LINKED ISSUE #1637: Issues continuing git-town sync
Ran into issues lately where after a sync hit a merge conflict, if I run `git town continue` after resolving conflicts but not committing, it doesn't push the branch (looks like its auto-aborting but I'm not sure).

Committing and then running `git town continue` works as expected. 

I'll try to get more details the next time this happens

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #1637 

Removed the steps for changing directory given that the go process does not actually change the directory of the calling process. As part of this cleaned up an unused test function.

### CODE REVIEW

File: test/data_table.go [hunk 1]
```diff
@@ -81,10 +80,6 @@ func (table *DataTable) Expand(rootDir string, localRepo *Repo, remoteRepo *Repo
 				templateOnce.Do(func() { templateRE = regexp.MustCompile(`\{\{.*?\}\}`) })
 				match := templateRE.FindString(cell)
 				switch {
-				case match == "{{ root folder }}":
```
Comment (kevgo) [inline test/data_table.go:84]: You can now remove the `rootDir` argument for this function.

File: features/hack/on_feature_branch/in_uncommitted_subfolder.feature [hunk 2]
```diff
@@ -19,17 +19,14 @@ Feature: git town-hack: starting a new feature from a new subfolder on the main
...
+      |             | git add -A                   |
       |             | git stash                    |
       |             | git checkout main            |
       | main        | git rebase origin/main       |
       |             | git branch new-feature main  |
       |             | git checkout new-feature     |
       | new-feature | git stash pop                |
-      | <none>      | cd {{ folder "new_folder" }} |
     And I end up on the "new-feature" branch
-    And I am in the project root folder
```
Comment (kevgo) [inline features/hack/on_feature_branch/in_uncommitted_subfolder.feature:32]: Shouldn't the tests still verify/specify in which folder we end up after running Git Town? I think that's important user-visible behavior.

**Review by kevgo — APPROVED** [review]
Thanks for this elegant fix! What do you think about leaving the specifications in which folder the user ends up?

File: features/hack/on_feature_branch/in_uncommitted_subfolder.feature (hunk 2 above)
Comment (charlierudolph) [reply features/hack/on_feature_branch/in_uncommitted_subfolder.feature:32]: I'm uncertain if that is worth doing. We could certainly add it but then I think we need to change a bit how we execute the changes.

Currently we have the step: [^I run "([^"]+)" in the "([^"]+)" folder$](https://github.com/git-town/git-town/blob/master/test/steps.go#L238) which runs a command in a dedicated folder vs changing a working dir and then executing it. If we want to switch to that, I think we just need to break that step up.

File: test/data_table.go (hunk 1 above)
Comment (charlierudolph) [reply test/data_table.go:84]: Nice catch, done

File: src/git/prod_repo.go [hunk 3]
```diff
@@ -70,3 +73,18 @@ func (r *ProdRepo) RemoveOutdatedConfiguration() error {
 	}
 	return nil
 }
+
+func (r *ProdRepo) NavigateToRootIfNecessary() error {
+	currentDirectory, err := os.Getwd()
+	if err != nil {
+		return fmt.Errorf("cannot get current working directory: %v", err)
```
Comment (kevgo) [inline src/git/prod_repo.go:80]: The rest of the codebase wraps errors with `%w` instead of `%v`.

**Review by kevgo — APPROVED** [review]
Very cool! Feel free to🚀 after the last comment is addressed.

**Comment by charlierudolph** [comment]
This was shipped via git-town but I was missing my github token

**Pull request closed without merging**